haystack/utils/augment_squad.py

```diff
@@ -0,0 +1,178 @@
+                continue
+            if random.random() < replace_probability:
+                new_text.append(random.choice(possible_words[1:]))
+            else:
+                new_text.append(possible_words[0])
+        new_texts.append(" ".join(new_text))
+    return new_texts
+
+def augment_squad(model: BertForMaskedLM, tokenizer: BertTokenizer, squad_path: Path, output_path: Path, glove_path: Path = Path("glove.txt"), multiplication_factor: int = 20, word_possibilities: int = 20, replace_probability: float = 0.4):
+    word_id_mapping, id_word_mapping, vectors = load_glove(glove_path)
+
+    with open(squad_path, "r") as f:
+        squad = json.load(f)
+        
+    topics = []
+
+    for topic in tqdm(squad["data"]):
+        paragraphs = []
+        for paragraph in topic["paragraphs"]:
+            context = paragraph["context"]
+            contexts = augment(word_id_mapping, id_word_mapping, vectors, model, tokenizer, context, multiplication_factor=multiplication_factor, word_possibilities=word_possibilities, replace_probability=replace_probability)
+            qas = []
+            for qa in paragraph["qas"]:
+                question = qa["question"]
+                question = augment(word_id_mapping, id_word_mapping, vectors, model, tokenizer, question, multiplication_factor=multiplication_factor, word_possibilities=word_possibilities, replace_probability=replace_probability)
+                qas_ = []
+                for question_ in question:
+                    new_qa = deepcopy(qa)
+                    new_qa["question"] = question_
+                    qas_.append(new_qa)
+                qas += qas_
+            paragraphs_ = []
+            for context in contexts:
+                new_paragraph = deepcopy(paragraph)
+                new_paragraph["context"] = context
+                new_paragraph["qas"] = qas
+                paragraphs_.append(new_paragraph)
+            paragraphs += paragraphs_
+        topic["paragraphs"] = paragraphs
+        topics.append(topic)
+    squad["topics"] = topics
+    with open(output_path, "w") as f:
+        json.dump(squad, f)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--squad_path", type=Path, required=True, help="Path to the squad json file")
+    parser.add_argument("--output_path", type=Path, required=True, help="Path to save augmented dataaset")
+    parser.add_argument("--multiplication_factor", type=int, default=4, help="Factor by which dataset size is multiplied")
+    parser.add_argument("--word_possibilities", type=int, default=5, help="Number of possible words to choose from when replacing a word")
+    parser.add_argument("--replace_probability", type=float, default=0.4, help="Probability of replacing a word")
+    parser.add_argument("--glove_path", type=Path, default="glove.txt", help="Path to the glove file")
+
+    model = BertForMaskedLM.from_pretrained("bert-base-uncased")
```

Comment:
Let's pass the model/tokenizer name as an argument to the script as well.

Reply:
I have added these arguments.